### PR TITLE
Add Yoga

### PR DESCRIPTION
**Pull request by hedtke**
Specify library name and version:  **yoga/2.0.0**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/yoga/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    name = "yoga"
+    description = "cross-platform layout engine which implements Flexbox"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://yogalayout.com/"
+    topics = ("layout engine", "flexbox")
+    package_type = "static-library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
```
Comment (RubenRBS) [inline recipes/yoga/all/conanfile.py:21]: If this is a static library and the shared option is disabled in the validate method, there's no reason to define it at all iirc, so feel free to drop it :)

**Review by RubenRBS — COMMENTED** [review]
Nice! Thanks a lot for your contribution. Other than the `shared` option comment, the rest looks good to me!

File: recipes/yoga/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        apply_conandata_patches(self)
+        tc = CMakeToolchain(self)
+        tc.generate()
+        tc = CMakeDeps(self)
+        tc.generate()
```
Comment (uilianries) [inline recipes/yoga/all/conanfile.py:68]:
```suggestion
```

The Yoga project has no dependencies, so it does not need CMakeDeps 😄

**Review by uilianries — APPROVED** [review]
LGTM

**Review by RubenRBS — APPROVED** [review]

**Pull request merged**